[openusd_internal] Fix operator== infinite recusion in C++20 mode

Something to do with "rewritten comparison operators" causes this
function to be a self-call in C++20, which means infinite recursion
and crash.

We should probably check out this problem in more detail, use a C++20
pre-processor guard (instead of if-0), and submit this upstream.


--- pxr/base/tf/weakPtrFacade.h
+++ pxr/base/tf/weakPtrFacade.h
@@ -135,10 +135,12 @@ public:
         return !(*this == p);
     }
 
+#if 0  // C++20 infinite recursion
     template <class T>
     friend bool operator == (const TfRefPtr<T>& p1, Derived const &p2) {
         return p2 == p1;
     }
+#endif
 
     template <class T>
     friend bool operator != (const TfRefPtr<T>& p1, Derived const &p2) {
